Scarecrow/field issue

### Is there an existing issue for this?

- [x] I have searched the existing issues.

### Are you using the latest MineColonies Version?

- [x] I am running the latest **beta/release** version of MineColonies for my Minecraft version.
I am also running the latest versions of other mods that are part of my problem.


### Did you check on the Wiki? or ask on Discord?

- [x] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### What were you playing at the time? Were you able to reproduce it in both settings?

- [x] Single Player
- [ ] Multi Player

### Minecraft Version

1.21

### MineColonies Version

1.1.1260

### Structurize Version

1.0.805

### Related Mods and their Versions

1. Forge: Sorry, idk how to check
2: DO: 1.0.228
3: BlockUI: 1.0.209

### Current Behavior

okayy so the scarecrows lose their assigned seed, thus being unallocated from farms, and whem i try to set them again, game chrases out instantly, giving this: "The game crashed: mousepressed event for bo screen
Error: java.lang.NullPointerException: Cannot invoke "com.minecolonies.core.colony.buildingextensions.FarmField.getSeed()" because "this.farmField" is null"

Now, I can break the scarecrows, and seemingly the issue disappears, can reassign them to each farmer, but after some time of minding my own business, I see that there is nothing going on on the fields, since the scarecrows got effed again.

aaand the logs: https://gist.github.com/c16kum/17f72be3948293d2f089a6918a96e344

### Expected Behavior

Well I think yall know that

### Reproduction Steps

Welp I have wrote it already, but I think csarecrows lose their seeds when I leave the game, but just a guess.

### Logs

https://gist.github.com/c16kum/17f72be3948293d2f089a6918a96e344

### Anything else?

_No response_

### Footer

---
 ## Viewers
  *  Add a 👍 reaction to the bug report if you are also affected. This helps the bug report become more visible without cluttering the comments.
  *  ⚠️ Only add a comment if you have new insights or background information not already mentioned. Off-topic or "+1" comments will be deleted to keep the discussion focused.

I recently updated the following modpacks to the following versions:

- MineColonies==1.1.1262
- Structurize==1.0.807
- Journeymap-NeoForge==1.21.1-6.0.0-beta.54+neoforge
- Domum Ornamentum==1.0.228
- Ferrite Core==7.0.3

Minecraft version: 1.21.1
NeoForge version: 21.1.219


Console log dump:
```
[07:37:53] [Render thread/ERROR] [minecraft/Minecraft]: Reported exception thrown!
net.minecraft.ReportedException: MousePressed event for BO screen
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.gui.screens.Screen.wrapScreenError(Screen.java:456) ~[client-1.21.1-20240808.144430-srg.jar%23244!/:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:computing_frames,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:journeymap.mixins.json:client.ScreenAccessor from mod journeymap,pl:mixin:APP:clickadv.mixins.json:ScreenMixin from mod clickadv,pl:mixin:A,pl:runtimedistcleaner:A}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.MouseHandler.onPress(MouseHandler.java:95) ~[client-1.21.1-20240808.144430-srg.jar%23244!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.MouseHandler.lambda$setup$4(MouseHandler.java:202) ~[client-1.21.1-20240808.144430-srg.jar%23244!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.BlockableEventLoop.execute(BlockableEventLoop.java:98) ~[client-1.21.1-20240808.144430-srg.jar%23244!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.MouseHandler.lambda$setup$5(MouseHandler.java:202) ~[client-1.21.1-20240808.144430-srg.jar%23244!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at MC-BOOTSTRAP/org.lwjgl.glfw@3.3.3+5/org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:43) ~[lwjgl-glfw-3.3.3.jar%23109!/:build 5] {}
	at MC-BOOTSTRAP/org.lwjgl@3.3.3+5/org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.3.3.jar%23121!/:build 5] {}
	at MC-BOOTSTRAP/org.lwjgl.glfw@3.3.3+5/org.lwjgl.glfw.GLFW.glfwPollEvents(GLFW.java:3438) ~[lwjgl-glfw-3.3.3.jar%23109!/:build 5] {re:mixin}
	at TRANSFORMER/minecraft@1.21.1/com.mojang.blaze3d.systems.RenderSystem.pollEvents(RenderSystem.java:134) ~[client-1.21.1-20240808.144430-srg.jar%23244!/:?] {re:mixin,pl:runtimedistcleaner:A,re:classloading,pl:mixin:APP:embeddium.mixins.json:workarounds.event_loop.RenderSystemMixin from mod embeddium,pl:mixin:A,pl:runtimedistcleaner:A}
	at TRANSFORMER/minecraft@1.21.1/com.mojang.blaze3d.systems.RenderSystem.flipFrame(RenderSystem.java:147) ~[client-1.21.1-20240808.144430-srg.jar%23244!/:?] {re:mixin,pl:runtimedistcleaner:A,re:classloading,pl:mixin:APP:embeddium.mixins.json:workarounds.event_loop.RenderSystemMixin from mod embeddium,pl:mixin:A,pl:runtimedistcleaner:A}
	at TRANSFORMER/minecraft@1.21.1/com.mojang.blaze3d.platform.Window.updateDisplay(Window.java:292) ~[client-1.21.1-20240808.144430-srg.jar%23244!/:?] {re:mixin,pl:runtimedistcleaner:A,re:classloading,pl:mixin:APP:embeddium.mixins.json:workarounds.context_creation.WindowMixin from mod embeddium,pl:mixin:A,pl:runtimedistcleaner:A}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.Minecraft.runTick(Minecraft.java:1217) ~[client-1.21.1-20240808.144430-srg.jar%23244!/:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:autowalk.mixins.json:MinecraftMixin from mod autowalk,pl:mixin:APP:autowalk.mixins.json:ScreenMixin from mod autowalk,pl:mixin:APP:journeymap.mixins.json:client.MinecraftMixin from mod journeymap,pl:mixin:APP:modernfix-modernfix.mixins.json:feature.measure_time.MinecraftMixin_Forge from mod modernfix,pl:mixin:APP:modernfix-modernfix.mixins.json:feature.measure_time.MinecraftMixin from mod modernfix,pl:mixin:APP:modernfix-modernfix.mixins.json:bugfix.world_leaks.MinecraftMixin from mod modernfix,pl:mixin:APP:modernfix-modernfix.mixins.json:perf.dedicated_reload_executor.MinecraftMixin from mod modernfix,pl:mixin:APP:memorysettings.mixins.json:MinecraftMixin from mod memorysettings,pl:mixin:APP:embeddium.mixins.json:core.MinecraftClientMixin from mod embeddium,pl:mixin:APP:embeddium.mixins.json:core.render.MinecraftAccessor from mod embeddium,pl:mixin:APP:modernfix-modernfix.mixins.json:feature.remove_telemetry.MinecraftMixin_Telemetry from mod modernfix,pl:mixin:A,pl:runtimedistcleaner:A}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.Minecraft.run(Minecraft.java:807) ~[client-1.21.1-20240808.144430-srg.jar%23244!/:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:autowalk.mixins.json:MinecraftMixin from mod autowalk,pl:mixin:APP:autowalk.mixins.json:ScreenMixin from mod autowalk,pl:mixin:APP:journeymap.mixins.json:client.MinecraftMixin from mod journeymap,pl:mixin:APP:modernfix-modernfix.mixins.json:feature.measure_time.MinecraftMixin_Forge from mod modernfix,pl:mixin:APP:modernfix-modernfix.mixins.json:feature.measure_time.MinecraftMixin from mod modernfix,pl:mixin:APP:modernfix-modernfix.mixins.json:bugfix.world_leaks.MinecraftMixin from mod modernfix,pl:mixin:APP:modernfix-modernfix.mixins.json:perf.dedicated_reload_executor.MinecraftMixin from mod modernfix,pl:mixin:APP:memorysettings.mixins.json:MinecraftMixin from mod memorysettings,pl:mixin:APP:embeddium.mixins.json:core.MinecraftClientMixin from mod embeddium,pl:mixin:APP:embeddium.mixins.json:core.render.MinecraftAccessor from mod embeddium,pl:mixin:APP:modernfix-modernfix.mixins.json:feature.remove_telemetry.MinecraftMixin_Telemetry from mod modernfix,pl:mixin:A,pl:runtimedistcleaner:A}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.main.Main.main(Main.java:230) ~[client-1.21.1-20240808.144430-srg.jar%23244!/:?] {re:mixin,pl:runtimedistcleaner:A,re:classloading,pl:mixin:APP:experimentalist.mixins.json:init.ClientMainMixin from mod experimentalist,pl:mixin:A,pl:runtimedistcleaner:A}
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?] {}
	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[?:?] {}
	at MC-BOOTSTRAP/fml_loader@4.0.42/net.neoforged.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:136) ~[loader-4.0.42.jar%23161!/:4.0] {}
	at MC-BOOTSTRAP/fml_loader@4.0.42/net.neoforged.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:124) ~[loader-4.0.42.jar%23161!/:4.0] {}
	at MC-BOOTSTRAP/fml_loader@4.0.42/net.neoforged.fml.loading.targets.CommonClientLaunchHandler.runService(CommonClientLaunchHandler.java:32) ~[loader-4.0.42.jar%23161!/:4.0] {}
	at MC-BOOTSTRAP/fml_loader@4.0.42/net.neoforged.fml.loading.targets.CommonLaunchHandler.lambda$launchService$4(CommonLaunchHandler.java:118) ~[loader-4.0.42.jar%23161!/:4.0] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) [modlauncher-11.0.5.jar%23166!/:?] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) [modlauncher-11.0.5.jar%23166!/:?] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) [modlauncher-11.0.5.jar%23166!/:?] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.Launcher.run(Launcher.java:103) [modlauncher-11.0.5.jar%23166!/:?] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.Launcher.main(Launcher.java:74) [modlauncher-11.0.5.jar%23166!/:?] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) [modlauncher-11.0.5.jar%23166!/:?] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) [modlauncher-11.0.5.jar%23166!/:?] {}
	at cpw.mods.bootstraplauncher@2.0.2/cpw.mods.bootstraplauncher.BootstrapLauncher.run(BootstrapLauncher.java:210) [bootstraplauncher-2.0.2.jar:?] {}
	at cpw.mods.bootstraplauncher@2.0.2/cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:69) [bootstraplauncher-2.0.2.jar:?] {}
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?] {}
	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[?:?] {}
	at io.github.zekerzhayard.forgewrapper.installer.Main.main(Main.java:67) [ForgeWrapper-prism-2025-12-07.jar:prism-2025-12-07] {}
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:115) [NewLaunch.jar:?] {}
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129) [NewLaunch.jar:?] {}
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70) [NewLaunch.jar:?] {}
Caused by: java.lang.NullPointerException: Cannot invoke "com.minecolonies.core.colony.buildingextensions.FarmField.getSeed()" because "this.farmField" is null
	at TRANSFORMER/minecolonies@1.1.1262-1.21.1-snapshot/com.minecolonies.core.client.gui.containers.WindowField.selectSeed(WindowField.java:146) ~[minecolonies-1.1.1262-1.21.1-snapshot.jar%23280!/:1.1.1262-1.21.1-snapshot] {re:classloading,pl:runtimedistcleaner:A}
	at TRANSFORMER/minecolonies@1.1.1262-1.21.1-snapshot/com.minecolonies.core.client.gui.AbstractWindowSkeleton.lambda$registerButton$2(AbstractWindowSkeleton.java:120) ~[minecolonies-1.1.1262-1.21.1-snapshot.jar%23280!/:1.1.1262-1.21.1-snapshot] {re:classloading}
	at TRANSFORMER/minecolonies@1.1.1262-1.21.1-snapshot/com.minecolonies.core.client.gui.AbstractWindowSkeleton.onButtonClicked(AbstractWindowSkeleton.java:194) ~[minecolonies-1.1.1262-1.21.1-snapshot.jar%23280!/:1.1.1262-1.21.1-snapshot] {re:classloading}
	at TRANSFORMER/blockui@1.0.209-1.21.1/com.ldtteam.blockui.controls.Button.handleClick(Button.java:115) ~[blockui-1.0.209-1.21.1.jar%23251!/:1.0.209-1.21.1] {re:classloading}
	at TRANSFORMER/blockui@1.0.209-1.21.1/com.ldtteam.blockui.Pane.click(Pane.java:597) ~[blockui-1.0.209-1.21.1.jar%23251!/:1.0.209-1.21.1] {re:classloading}
	at TRANSFORMER/blockui@1.0.209-1.21.1/com.ldtteam.blockui.views.View.mouseEventProcessor(View.java:438) ~[blockui-1.0.209-1.21.1.jar%23251!/:1.0.209-1.21.1] {re:classloading}
	at TRANSFORMER/blockui@1.0.209-1.21.1/com.ldtteam.blockui.views.View.mouseClickableEventHandler(View.java:391) ~[blockui-1.0.209-1.21.1.jar%23251!/:1.0.209-1.21.1] {re:classloading}
	at TRANSFORMER/blockui@1.0.209-1.21.1/com.ldtteam.blockui.views.View.click(View.java:234) ~[blockui-1.0.209-1.21.1.jar%23251!/:1.0.209-1.21.1] {re:classloading}
	at TRANSFORMER/blockui@1.0.209-1.21.1/com.ldtteam.blockui.BOScreen.mouseClicked(BOScreen.java:196) ~[blockui-1.0.209-1.21.1.jar%23251!/:1.0.209-1.21.1] {re:classloading}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.MouseHandler.lambda$onPress$0(MouseHandler.java:98) ~[client-1.21.1-20240808.144430-srg.jar%23244!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.gui.screens.Screen.wrapScreenError(Screen.java:451) ~[client-1.21.1-20240808.144430-srg.jar%23244!/:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:computing_frames,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:journeymap.mixins.json:client.ScreenAccessor from mod journeymap,pl:mixin:APP:clickadv.mixins.json:ScreenMixin from mod clickadv,pl:mixin:A,pl:runtimedistcleaner:A}
	... 34 more
```

--- a/src/main/java/com/minecolonies/api/colony/managers/interfaces/IRegisteredStructureManager.java
+++ b/src/main/java/com/minecolonies/api/colony/managers/interfaces/IRegisteredStructureManager.java
@@ -1,6 +1,7 @@
 package com.minecolonies.api.colony.managers.interfaces;
 
 import com.minecolonies.api.colony.IColony;
+import com.minecolonies.api.colony.buildingextensions.registry.BuildingExtensionRegistries;
 import com.minecolonies.api.colony.buildings.IBuilding;
 import com.minecolonies.api.colony.buildings.IMysticalSite;
 import com.minecolonies.api.colony.buildings.workerbuildings.ITownHall;
@@ -265,4 +266,11 @@ public interface IRegisteredStructureManager extends ICommonRegisteredStructureM
      * @return the building extension or null.
      */
     @Nullable IBuildingExtension getMatchingBuildingExtension(IBuildingExtension.ExtensionId extensionId);
+
+    /**
+     * Add a building extension if it's missing.
+     * @param buildingExtensionEntry the entry to create the extension from.
+     * @param pos the pos it's at.
+     */
+    void addBuildingExtensionIfMissing(BuildingExtensionRegistries.BuildingExtensionEntry buildingExtensionEntry, BlockPos pos, final Player player));
 }

--- a/src/main/java/com/minecolonies/api/colony/managers/interfaces/IRegisteredStructureManager.java
+++ b/src/main/java/com/minecolonies/api/colony/managers/interfaces/IRegisteredStructureManager.java
@@ -272,5 +272,5 @@ public interface IRegisteredStructureManager extends ICommonRegisteredStructureM
      * @param buildingExtensionEntry the entry to create the extension from.
      * @param pos the pos it's at.
      */
-    void addBuildingExtensionIfMissing(BuildingExtensionRegistries.BuildingExtensionEntry buildingExtensionEntry, BlockPos pos, final Player player));
+    void addBuildingExtensionIfMissing(BuildingExtensionRegistries.BuildingExtensionEntry buildingExtensionEntry, BlockPos pos, final Player player);
 }

--- a/src/main/java/com/minecolonies/core/blocks/BlockScarecrow.java
+++ b/src/main/java/com/minecolonies/core/blocks/BlockScarecrow.java
@@ -99,6 +99,11 @@ public class BlockScarecrow extends AbstractBlockMinecoloniesDefault<BlockScarec
             }
         }
 
+        final IColony iColony = IColonyManager.getInstance().getIColony(worldIn, pos);
+        if (iColony != null)
+        {
+            iColony.getServerBuildingManager().addBuildingExtensionIfMissing(BuildingExtensionRegistries.farmField.get(), pos, player);
+        }
         // This must succeed in Remote to stop more right click interactions like placing blocks
         return ItemInteractionResult.SUCCESS;
     }

--- a/src/main/java/com/minecolonies/core/colony/managers/RegisteredStructureManager.java
+++ b/src/main/java/com/minecolonies/core/colony/managers/RegisteredStructureManager.java
@@ -873,7 +873,7 @@ public class RegisteredStructureManager implements IRegisteredStructureManager
     public void addBuildingExtensionIfMissing(final BuildingExtensionRegistries.BuildingExtensionEntry buildingExtensionEntry, final BlockPos pos, final Player player)
     {
         buildingExtensions.computeIfAbsent(new IBuildingExtension.ExtensionId(pos, buildingExtensionEntry), (id) -> {
-            new ColonyViewBuildingExtensionsUpdateMessage(colony, buildingExtensions.values()).sendToPlayer(player);
+            new ColonyViewBuildingExtensionsUpdateMessage(colony, buildingExtensions.values()).sendToPlayer((ServerPlayer) player);
             markBuildingExtensionsDirty();
             return buildingExtensionEntry.produceExtension(pos);
         });

--- a/src/main/java/com/minecolonies/core/colony/managers/RegisteredStructureManager.java
+++ b/src/main/java/com/minecolonies/core/colony/managers/RegisteredStructureManager.java
@@ -7,6 +7,7 @@ import com.minecolonies.api.blocks.AbstractBlockHut;
 import com.minecolonies.api.colony.IAnimalData;
 import com.minecolonies.api.colony.ICitizenData;
 import com.minecolonies.api.colony.IColony;
+import com.minecolonies.api.colony.buildingextensions.registry.BuildingExtensionRegistries;
 import com.minecolonies.api.colony.buildings.*;
 import com.minecolonies.api.colony.buildings.registry.IBuildingDataManager;
 import com.minecolonies.api.colony.buildings.workerbuildings.ITownHall;
@@ -866,6 +867,16 @@ public class RegisteredStructureManager implements IRegisteredStructureManager
     public IBuildingExtension getMatchingBuildingExtension(final IBuildingExtension.ExtensionId extensionId)
     {
         return buildingExtensions.get(extensionId);
+    }
+
+    @Override
+    public void addBuildingExtensionIfMissing(final BuildingExtensionRegistries.BuildingExtensionEntry buildingExtensionEntry, final BlockPos pos, final Player player)
+    {
+        buildingExtensions.computeIfAbsent(new IBuildingExtension.ExtensionId(pos, buildingExtensionEntry), (id) -> {
+            new ColonyViewBuildingExtensionsUpdateMessage(colony, buildingExtensions.values()).sendToPlayer(player);
+            markBuildingExtensionsDirty();
+            return buildingExtensionEntry.produceExtension(pos);
+        });
     }
 
     @Override

--- a/src/main/java/com/minecolonies/core/event/EventHandler.java
+++ b/src/main/java/com/minecolonies/core/event/EventHandler.java
@@ -64,6 +64,7 @@ import net.minecraft.world.level.storage.loot.LootPool;
 import net.minecraft.world.level.storage.loot.entries.LootItem;
 import net.minecraft.world.level.storage.loot.predicates.LootItemRandomChanceCondition;
 import net.neoforged.bus.api.SubscribeEvent;
+import net.neoforged.neoforge.client.event.ClientPlayerNetworkEvent;
 import net.neoforged.neoforge.event.EventHooks;
 import net.neoforged.neoforge.event.LootTableLoadEvent;
 import net.neoforged.neoforge.event.RegisterCommandsEvent;
@@ -682,11 +683,13 @@ public class EventHandler
         {
             IColonyManager.getInstance().onWorldUnload((Level) event.getLevel());
         }
-        if (event.getLevel().isClientSide())
-        {
-            IColonyManager.getInstance().resetColonyViews();
-            Log.getLogger().info("Removed all colony views");
-        }
+    }
+
+    @SubscribeEvent
+    public static void onPlayerLogout(@NotNull final ClientPlayerNetworkEvent.LoggingOut event)
+    {
+        IColonyManager.getInstance().resetColonyViews();
+        Log.getLogger().info("Removed all colony views");
     }
 
     /**

